federation/pkg/federation-controller/secret/secret_controller_test.go

```diff
@@ -127,15 +127,21 @@ func TestSecretController(t *testing.T) {
 	assert.Equal(t, secret1.Namespace, updatedSecret.Namespace)
 	assert.True(t, secretsEqual(secret1, *updatedSecret),
 		fmt.Sprintf("expected: %v, actual: %v", secret1, *updatedSecret))
+	// Wait for the secret to be updated in the informer store.
+	err = WaitForSecretStoreUpdate(
+		secretController.secretFederatedInformer.GetTargetStore(),
+		cluster1.Name, types.NamespacedName{Namespace: secret1.Namespace, Name: secret1.Name}.String(),
+		updatedSecret, wait.ForeverTestTimeout)
```

Comment:
I will prefer keeping this as is. Dont want to introduce more flakiness because of not enough wait :)